### PR TITLE
Uploading the ISC Service User Guide

### PR DESCRIPTION
**Pull request by VenkyOS**

### CODE REVIEW

**Review by andriytk — CHANGES_REQUESTED** [review]
A lot of code examples are given as quotations, please fix.

**Comment by VenkyOS** [comment]
> A lot of code examples are given as quotations, please fix.

Can you please elaborate on what needs to be done?

**Comment by andriytk** [comment]
> ... elaborate on what needs to be done?

Fix the MD-formatting, I guess?

**Comment by VenkyOS** [comment]
> > ... elaborate on what needs to be done?
> 
> Fix the MD-formatting, I guess?

In .rst format, this is the way snippets are represented.

File: doc/ISC_Service_User_Guide.rst [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+ int comp(struct m0_buf *args, struct m0_buf *out,
+
+          struct m0_isc_comp_private *comp_data, int *rc)
+
+All relevant library APIs shall be prepared with a wrapper confining to this signature. Let libarray be the library we intend to link with Motr, with following APIs: arr_max(), arr_min(), arr_histo().
+
+Registering APIs
+=================
+
+motr_lib_init() links all the APIs. Here is an example code (please see iscservice/isc.h for more details):
```
Comment (huanghua78) [inline doc/ISC_Service_User_Guide.rst:22]: Please use:
```
.. code-block:: C
```
to embed C code

Comment (huanghua78) [reply doc/ISC_Service_User_Guide.rst:21]: Please see my commits.

**Review by huanghua78 — APPROVED** [review]

**Review by andriytk — APPROVED** [review]
Looks fine now.

**Pull request merged**